Deprecates X-XSS-Protection and put it behind feature flag. (#36881)
keycloak/keycloak: docs/documentation/upgrading/topics/changes/changes-26_2_0.adoc

> = Removal of the `X-XSS-Protection` header

Reviewer: Is it possible to add this to the release notes for 26.2.0 instead of upgrading guide? AFAIK upgrading guide may contain only things, which somehow affect backwards compatibility and require some action from the Keycloak administrators/users. But this is not the case if I understand correctly?

Reviewer 2: This still might affect some users if they re-use this header, I think the changelog is the perfect place for it. Putting it in the release notes would just clutter them, I'd rather reserve it for a nice overview of features.

Reviewer 2: Resolving this as per https://github.com/keycloak/keycloak/pull/36881#issuecomment-2666546172

> Because the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection[`X-XSS-Protection` header] is no longer supported by any of the user agents that are supported by Keycloak it has been removed. This header was a feature of Internet Explorer, Chrome and Safari that stopped pages from loading when they detected reflected cross-site scripting (XSS) attacks.

Reviewer 3: ```suggestion
Because the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection[`X-XSS-Protection` header] is no longer supported by any user agents that are supported by Keycloak, it has been removed. This header was a feature of Internet Explorer, Chrome, and Safari that stopped pages from loading when they detected reflected cross-site scripting (XSS) attacks.
```

= Removal of XSS Header

We have removed the X-XSS-Protection response header from Keycloak, this header was a feature of Internet Explorer,
Chrome and Safari that stopped pages from loading when they detected reflected cross-site scripting (XSS) attacks.

> Since, this header is no longer supported in any of the user agents supported by Keycloak and has been replaced by

Reviewer 3: ```suggestion
Since this header is no longer supported in any user agents that are supported by Keycloak and has been replaced by
```

> Since, this header is no longer supported in any of the user agents supported by Keycloak and has been replaced by
> Content-Security-Policy. More details about the header can be found https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection[here]

Reviewer 3: ```suggestion
Since this header is no longer supported in any of the user agents supported by Keycloak, it has been replaced by
Content-Security-Policy. For more details, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection[X-XSS-Protection].
```
